zerver/decorator.py

```diff
@@ -449,6 +449,15 @@ def _wrapped_view_func(request: HttpRequest, *args: Any, **kwargs: Any) -> HttpR
         return add_logging_data(view_func)(request, *args, **kwargs)
     return _wrapped_view_func  # type: ignore # https://github.com/python/mypy/issues/1927
 
+def require_server_admin_access(view_func: ViewFuncT) -> ViewFuncT:
+    @zulip_login_required
+    @wraps(view_func)
+    def _wrapped_view_func(request: HttpRequest, user_profile: UserProfile, *args: Any, **kwargs: Any) -> HttpResponse:
+        if not user_profile.is_staff:
+            raise JsonableError(_("Must be an server administrator"))
+        return view_func(request, user_profile, *args, **kwargs)
+    return _wrapped_view_func  # type: ignore # https://github.com/python/mypy/issues/1927
```

Comment:
This should come with some tests in `test_decorator.py`.  

Also, I'd call it `require_server_admin_api` -- since the point is it's for API/JSON type views.